include/eve/traits/common_type.hpp

```diff
@@ -63,6 +62,15 @@ namespace eve::detail
     friend constexpr auto operator+(no_common_type, U) { return no_common_type{}; }
   };
 
+  template <typename T>
+  struct is_kumi_tuple : std::false_type {};
+
+  template <typename ...Ts>
+  struct is_kumi_tuple<kumi::tuple<Ts...>> : std::true_type {};
+
+  template <typename T>
+  concept is_kumi_tuple_v = is_kumi_tuple<T>::value;
```

Comment:
why not using kumi::product_type ?

Reply:
It's not the same check.  I need to prefer a product type with semantics to kumi tuple